src/authentication/authenticator.ts

```diff
@@ -67,6 +67,10 @@ export class Authenticator {
       return Promise.resolve(token);
     }
 
+    if (hasTokenExpired) {
```

Comment:
This fails on first load, if token doesn't already exist in the localStorage. Get returns `null` and `TokenStorage.hasExpired(token)` returns `true`. Better is to also check on actual existence of the key:
```suggestion
    if (token && hasTokenExpired) {
```

Reply:
@YannickRe Thanks! I made that change. Let me know if there's anything else.